betatest/tests.py

```diff
@@ -0,0 +1,158 @@
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Tests for the betatest app
+"""
+
+# Imports #####################################################################
+
+import re
+
+from django.core import mail
+from django.core.urlresolvers import reverse
+from django.test import TestCase
+
+from betatest.models import BetaTestApplication
+from email_verification.models import EmailVerification
+
+
+# Tests #######################################################################
+
+class BetaTestApplicationTestCase(TestCase):
+    """
+    Tests for beta test applications.
+    """
+    url = reverse('beta:register')
+
+    def setUp(self):
+        self.application = {
+            'subdomain': 'hogwarts',
+            'instance_name': 'Hogwarts',
+            'full_name': 'Albus Dumbledore',
+            'username': 'albus',
+            'email': 'albus.dumbledore@hogwarts.edu',
+            'public_contact_email': 'support@hogwarts.edu',
+            'password': 'gryffindor',
+            'password_confirmation': 'gryffindor',
+            'project_description': 'Online courses in Witchcraft and Wizardry',
+            'accept_terms': 'on',
+        }
+
+    def assert_registration_succeeds(self, application):
+        """
+        Assert that the given application form data creates new user, profile
+        and registration instances, sends email verification messages, and
+        displays a success message.
+        """
+        response = self.client.post(self.url, application, follow=True)
+        self.assertContains(response, 'Thank you for registering')
+
+        # An application, user and profile should have been created
+        application = BetaTestApplication.objects.get()
+        user = application.user
+        profile = user.profile
+
+        # Check the application fields
+        for application_field in ('subdomain',
+                                  'instance_name',
+                                  'public_contact_email',
+                                  'project_description'):
+            self.assertEqual(getattr(application, application_field),
+                             self.application[application_field])
+        self.assertEqual(application.subscribe_to_updates,
+                         bool(self.application.get('subscribe_to_updates')))
+
+        # Check the user fields
+        for user_field in ('username', 'email'):
+            self.assertEqual(getattr(user, user_field),
+                             self.application[user_field])
+        self.assertTrue(user.check_password(self.application['password']))
+
+        # Check the profile fields
+        self.assertEqual(profile.full_name, self.application['full_name'])
+
+        # Test email verification flow
+        self.assertEqual(len(mail.outbox), 2)
```

Comment:
@omarkhan Should we check if email is *unverified* before clicking the links below?